app/modules/gpio.c

```diff
@@ -87,59 +76,55 @@ static int lgpio_trig( lua_State* L )
 // Lua: mode( pin, mode, pullup )
 static int lgpio_mode( lua_State* L )
 {
-  unsigned mode, pullup = FLOAT;
-  unsigned pin;
+  unsigned pin = luaL_checkinteger( L, 1 );
+  unsigned mode = luaL_checkinteger( L, 2 );
+  unsigned pullup = luaL_optinteger( L, 3, FLOAT );
+  
+  luaL_argcheck(L, pin< NUM_GPIO && (mode!=INTERRUPT || pin>0), 1, "Invalid pin");
```

Comment:
Any good reason why you're not using `platform_gpio_exists(pin)` (or the `MOD_CHECK_ID(gpio,pin)` macro for it) here? GPIOs aren't necessarily contiguous (see ESP32 pinout for example).